crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -54,13 +58,29 @@ impl FormatNodeRule<Arguments> for FormatArguments {
                     };
                 }
                 args => {
-                    joiner
-                        .entries(
-                            // We have the parentheses from the call so the item never need any
-                            args.iter()
-                                .map(|arg| (arg, arg.format().with_options(Parentheses::Preserve))),
-                        )
-                        .nodes(item.keywords.iter());
+                    // Consider e.g. `f(*args, a=b, *args2, **kwargs)`, where args (`*args`,
+                    // `*args2`) and keywords (`a=b`, `**kwargs`) are interleaved
+                    let mut args = args.iter().peekable();
+                    let mut keywords = keywords.iter().peekable();
+                    loop {
```

Comment:
i unfortunately didn't find any pretty way to write this. itertool's `merge_by` also doesn't work due to different type

Reply:
Considerably more code but avoids `peekable`. Not sure if it easier to understand due to avoiding `map`

```rust
		let mut args = args.iter();
    let mut keywords = keywords.iter();

    let mut current_arg = args.next();
    let mut current_keyword = keywords.next();

    loop {
        match (current_arg, current_keyword) {
            (Some(arg), Some(keyword)) => {
                if arg.start() < keyword.start() {
                    joiner.entry(arg, &arg.format());
                    current_arg = args.next();
                } else {
                    joiner.entry(keyword, &keyword.format());
                    current_keyword = keywords.next();
                }
            }

            (None, Some(keyword)) => {
                for keyword in std::iter::once(keyword).chain(keywords) {
                    joiner.entry(keyword, &keyword.format());
                }

                break;
            }

            (Some(arg), None) => {
                for arg in std::iter::once(arg).chain(args) {
                    joiner.entry(arg, &arg.format());
                }

                break;
            }
            (None, None) => {
                break;
            }
        }
    }
}
```

Reply:
This will be nice and clean with your `ArgOrKeyword` abstraction :)